mate_menu/plugins/applications.py

```diff
@@ -1534,15 +1557,18 @@ def updateBoxes( self, menu_has_changed ):
                     else:
                         launcherNames.append(launcherName)
         except Exception as e:
-            print e
+            print(e)
 
         self.rebuildLock = False
 
     # Reload the menufiles from the filesystem
     def loadMenuFiles( self ):
-        self.menuFiles = []
-        for mainitems in [ "mate-applications.menu", "mate-settings.menu" ]:
-            self.menuFiles.append( Menu( mainitems) )
+        if len(self.menuFiles) == 0:
```

Comment:
I think this function is used to reload all menu items when you tell mate-menu to do it (right-click on menu button and choose "reload plugins"). Therefore, it should empty `self.menuFiles` first.

Looks like it can also be called when the menu itself changes (desktop apps are installed or removed), and when the icon theme changes.

Reply:
In class init function, already let self.menuFiles empty, so here maybe check it enough.

Reply:
But this is a reload, it implies that the old menu entries should be removed first...

Reply:
Rewrite code here, please test it again, thank you.